### ISSUE TITLE
Increasing loss but increasing precision

### ISSUE THREAD

**Issue by Paulito-7** [matterport/Mask_RCNN#453]
@waleedka 
First I would like to say thank you for this great implementation of Mask R-CNN!
I am currently working on my own dataset but I noticed a strange behavior when training.
I divided my training stage as follows, using a lr_decay in `model.py`

`model.train(train_dataset, val_dataset, 
                learning_rate=config.LEARNING_RATE, 
                epochs=50, 
                layers='heads')
model.train(train_dataset, val_dataset, 
                learning_rate=config.LEARNING_RATE/10, 
                epochs=70, 
                layers='4+')
model.train(train_dataset, val_dataset, 
                learning_rate=config.LEARNING_RATE/10, 
                epochs=90 
                layers='3+')
`
My `val_loss` goes up when training lower stages, but the result, when testing on some images, actually improves. 
Does it come from the way that are taken into account false positive / false negatives in the loss computation or is it something else? Have you ever met that matter?

Thank you !

**Comment by Paulito-7** [comment]
Thank you for your answer.
I will try to increase the size of my dataset and will keep you updated!

**Comment by Avi-avidan** [comment]
Hi Waleed,

I have made a few changes to model.py, let me know if you want to me to
push them -


   - added imgaug augmentations (lines 1234-1267 in the file attached)
      - CropAndPad
      - translate_percent
      - scale
      - rotate
      - shear
      - added 'Adam' Optimizer (lines 2153-2160)
   - added a change to 'build_fpn_mask_graph' so mask_shape is according to
   set in config.MASK_SHAPE (lines 2001, 2051, 951, 990-1000)
   - renamed to 'detections' output (inference only) so it would be easier
   to deploy in production (using tf.serving) (lines 2053-2054)


thanks for this repo. It is great.


Best,
Avi Avidan
0523385564

**Comment by fabioaraujopt** [comment]
what's the update?

**Comment by 1chimaruGin** [comment]
@Paulito-7 

update sir
